staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -78,6 +78,20 @@ var (
 		},
 		[]string{"verb", "resource", "subresource", "scope"},
 	)
+	// DroppedRequests is a number of requests dropped with 'Try again later' reponse"
+	DroppedRequests = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "apiserver_dropped_requests",
+			Help: "Number of requests dropped with 'Try again later' reponse",
+		},
+	)
+	// DroppedMutatingRequests is a number of mutating requests dropped with 'Try again later' reponse
+	DroppedMutatingRequests = prometheus.NewCounter(
```

Comment:
Why can't we have a single metric with label mutating set to true or false?
I think it would be much cleaner when looking into it.

Reply:
I was considering this, but I'm not sure how prometheus is implemented. It might have a lock per variable, not variable:label-set pair, which would make two variables slightly more efficient. If that's the case we gain a bit on contention by paying a bit on cleanness of code (but also not much), which is why I chose two variables approach.

Reply:
Personally I don't think it just cleanness of the code. If I don't know the metrics (and I will forget about it in few months probably), as soon as I will find the first one I will stop looking further. So i think it may be misleading for people using them.

That said, raising the lock contention here is a good point. But I'm not convinced how much effect this lock contention would have. Do we have some measurements? My personal opinion is that due to how apiserver is structures, those channels for tokens for max inflight requests, etc., the increase cost of lock contention will be negligible.

Reply:
I'll take your word on that.